### ISSUE TITLE
Upgrading to the new version Wandb seemed to cause a breakage of SHAP in my PyTorch code

### ISSUE THREAD

**Issue by isaacmg** [wandb/client#2140]
**Description**
Prior to upgrading to Wandb from 0.10.27 to 0.10.28 SHAP ` deep_explainer.shap_value(tensor)` ran without error. Now I receiving the below error: 

```
 File "/home/circleci/repo/flood_forecast/evaluator.py", line 184, in evaluate_model
    model, test_data, inference_params["datetime_start"]
  File "/home/circleci/repo/flood_forecast/explain_model_output.py", line 78, in deep_explain_model_summary_plot
    shap_values = deep_explainer.shap_values(background_tensor)
  File "/usr/local/lib/python3.7/site-packages/shap/explainers/_deep/__init__.py", line 124, in shap_values
    return self.explainer.shap_values(X, ranked_outputs, output_rank_order, check_additivity=check_additivity)
  File "/usr/local/lib/python3.7/site-packages/shap/explainers/_deep/deep_pytorch.py", line 163, in shap_values
    handles = self.add_handles(self.model, add_interim_values, deeplift_grad)
  File "/usr/local/lib/python3.7/site-packages/shap/explainers/_deep/deep_pytorch.py", line 76, in add_handles
    handles_list.extend(self.add_handles(child, forward_handle, backward_handle))
  File "/usr/local/lib/python3.7/site-packages/shap/explainers/_deep/deep_pytorch.py", line 79, in add_handles
    handles_list.append(model.register_backward_hook(backward_handle))
  File "/usr/local/lib/python3.7/site-packages/torch/nn/modules/module.py", line 690, in register_backward_hook
    raise RuntimeError("Cannot use both regular backward hooks and full backward hooks on a "
RuntimeError: Cannot use both regular backward hooks and full backward hooks on a single Module. Please
```

Once I downgraded back to 0.10.27 the problem went away. I'm guessing this is related to the way PyTorch models are being watched by Wandb in the new ver.

**Wandb features**
Using Weights and Biases with PyTorch.  

**How to reproduce**
Relevant code
https://github.com/AIStream-Peelout/flow-forecast/blob/6298161649c9331613988a449263df6925147222/flood_forecast/explain_model_output.py#L112

CircleCI reports
https://app.circleci.com/pipelines/github/AIStream-Peelout/flow-forecast/2930/workflows/f0ba4206-ee0e-465c-9c90-3c2ec7cc8092/jobs/12939 (with 0.10.27)
https://app.circleci.com/pipelines/github/AIStream-Peelout/flow-forecast/2929/workflows/799ee872-2c98-4ac1-a9fa-2506a45ffe51/jobs/12932 (with 0.10.28)

Wandb runs 
https://wandb.ai/igodfried/repo-flood_forecast/runs/y3y6ptx7/logs

**Environment**
- OS: Linux
- Environment: Colab, CircleCI
- Python Version: 3.7.4

**Comment by vanpelt** [comment]
Yep, we changed to full_backward_hooks in wandb.watch in the latest release.  This looks like we'll need to add an option to our `wandb.watch` call that let's tell us not to use full_hooks.

**Comment by vanpelt** [comment]
This was fixed in 0.10.31
